src/Operations.jl

```diff
@@ -18,6 +18,12 @@ import ...Pkg: pkg_server
 #########
 # Utils #
 #########
+
+const pkgs_precompile_suspended = Base.PkgId[]
```

Comment:
Doesn't this need to be `empty!`d somewhere?

Reply:
Specific packages are filtered out in the manifest update method. I'm not sure what action would warrant fully emptying it. Maybe at the start of a manual precomp invoke, which could serve as the means to ignore the suspension

Reply:
This empties `pkgs_precompile_suspended` as a means to retry precomp during manual calls

Reply:
Perhaps I am misunderstanding what this is used for, but if a package fails to precompile, why should I have to restart Julia in order to precompile it again (perhaps at a different version)?

Reply:
See line 139. The un-suspension happens when the package changes in the manifest

Reply:
Thanks. Wouldn't it be easier to just empty it everytime we enter `precompile`?

Reply:
That's what I just changed it to, but only in manual enters. 

We always enter if auto is enabled, the suspension is observed on a per-pkg basis, so we only want to empty! for the manual call.

Reply:
But consider if `Example@1.2.0` fails precompilation, and I add `Example@1.2.1`. Doesn't this mechanism prevent that from precompiling?

Reply:
Nevermind, you do empty it completely now.

Reply:
> Doesn't this mechanism prevent that from precompiling?

It's not the emptying that makes this work, it's that if you did change that package, it will be updated in the manifest, which will unsuspend that specific package.